frontend/src/components/taxons/EditTaxon.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button, Form, Input, Space } from "antd";
+import React, { useCallback, useState } from "react";
+import { requiredRules } from "../../constants";
+import { useAppDispatch, useAppSelector } from "../../hooks";
+import { selectAppInitialzed, selectTaxonsByID } from "../../selectors";
+import { useNavigate, useParams } from "react-router-dom";
+import { add, update } from "../../modules/taxons/actions";
+import AppPageHeader from "../AppPageHeader";
+import PageContent from "../PageContent";
+
+export interface Taxon {
```

Comment:
The type `Taxon` is already defined in `frontend_models.ts`. I would name this "EditableTaxon" to avoid confusion. Also, I don't think you need to export this interface, since it will only be used in this file.

Reply:
Another option would be to use Pick<> to define a new type from the existing Taxon definition:
`type EditableTaxon = Pick<Taxon, 'id'|'ncbi_id'|'name'>`
The only trouble is that you need id to be optional, so you would have to also use `Partial<>`, which makes all properties optional:
`type EditableTaxon = Partial<Pick<Taxon, 'id'|'ncbi_id'|'name'>>`

This is the point where some devs start to hate typescript :)

The advantage to doing it this way is that if the `Taxon` type is changed, and one of those properties no longer exists, then the code will break and the compiler will discover the problem for future us.